container/image.bzl

```diff
@@ -299,6 +307,11 @@ def _impl(
         legacy_run_behavior = ctx.attr.base[ImageInfo].legacy_run_behavior
         docker_run_flags = ctx.attr.base[ImageInfo].docker_run_flags
 
+    if ctx.attr.launcher:
+        if not file_map:
+            file_map = {}
+        file_map["/" + ctx.file.launcher.basename] = ctx.file.launcher
```

Comment:
Is is enough to just add the launcher file to validly use an arbitrary xx_binary target as the launcher
(i.e., what if the launcher file depends on other runfiles, like for a java_binary target)?
If not lets document it very well that the assumption is the launcher has no dependencies at all.
You should also clarify that the launcher needs to be compatible with the container OS (i.e., if you build the go_binary in Bazel running on mac, that cannot be used as a launcher attr).